src/utils.jl

```diff
@@ -228,20 +228,45 @@ macro cplwarn(code, message)
     end
 end
 
+function showprogress(progress, message = "")
+    print("$(round(Int, progress)*100)..$message")
+    return true
+end
+
+function dummyprogress(progress, message = "")
+    return true
+end
+
+function progressfunc_wrapper(
+    dfComplete::Cdouble,
+    pszMessage::Cstring,
+    pProgressArg::Ptr{Cvoid},
+)::Cint
```

Comment:
Can we make this an internal function and provide documentation for it? E.g. (based on the documentation for [GDALTermProgress](https://gdal.org/api/cpl.html#_CPPv416GDALTermProgressdPKcPv)),
```julia
"""
    _progresscallback(dfComplete::Cdouble, pszMessage::Cstring, pProgressArg::Ptr{Cvoid})::Cint

The progress callback to be passed to GDAL.

### Parameters
* `dfComplete` – completion ratio from 0.0 to 1.0.
* `pszMessage` – optional message (made available as an argument to the user-defined progress function).
* `pProgressArg` – callback data argument (i.e. user-defined progress function).
"""
```

Reply:
Resolved by https://github.com/yeesian/ArchGDAL.jl/pull/353/commits/24c1304555c48b8e4931b9755e3113f35e312a52